epochStart/metachain/rewardsV2_test.go

```diff
@@ -1127,31 +1132,139 @@ func TestNewRewardsCreatorV2_computeRewardsPer3200Nodes(t *testing.T) {
+	epochsInYear := big.NewInt(365)
 
-			sumRewards := big.NewInt(0)
-			for _, nodeInfoList := range nodesRewardInfo {
-				for _, nodeInfo := range nodeInfoList {
-					sumRewards.Add(sumRewards, big.NewInt(0).Mul(nodeInfo.fullRewards, big.NewInt(epochsInYear.Int64())))
+	rewardsForBlocks := big.NewInt(0).Div(totalRewardsFirstYear, epochsInYear)
+	baseStake := big.NewInt(0).Mul(big.NewInt(2500), smallesDivisionERD)
+	return SetupRewardsResult{
+		args, smallesDivisionERD, totalRewardsFirstYear, epochsInYear, rewardsForBlocks, baseStake,
+	}
+}
 
-					yearRewardsTimes100 := big.NewInt(0).Mul(nodeInfo.fullRewards, big.NewInt(epochsInYear.Int64()*factor))
-					roiInt := big.NewInt(0).Div(yearRewardsTimes100, big.NewInt(0).Add(tt.baseStake, tt.topupStake))
-					actualROI := float64(roiInt.Int64()) * 100 / float64(factor) * 0.9
+func computeRewardsAndDust(nbEligiblePerShard uint32, args SetupRewardsResult, topupStake *big.Int) (map[uint32][]*nodeRewardsData, *big.Int) {
+	vInfo := createDefaultValidatorInfo(nbEligiblePerShard, args.ShardCoordinator, args.NodesConfigProvider, 100, defaultBlocksPerShard)
+	dummyRwd, _ := NewRewardsCreatorV2(args.RewardsCreatorArgsV2)
+	nodesRewardInfo := dummyRwd.initNodesRewardsInfo(vInfo)
+	_, totalTopUpStake := setValuesInNodesRewardInfo(nodesRewardInfo, topupStake, tuStake)
+
+	totalEligibleStake, _ := big.NewInt(0).SetString("4000000"+"000000000000000000", 10)
 
-					diff := math.Abs(tt.expectedROI - actualROI)
-					require.True(t, diff < tolerance, fmt.Sprintf("expected %f, actual %f", tt.expectedROI, actualROI))
+	args.StakingDataProvider = &mock.StakingDataProviderStub{
+		GetTotalTopUpStakeEligibleNodesCalled: func() *big.Int {
+			return totalTopUpStake
+		},
+		GetTotalStakeEligibleNodesCalled: func() *big.Int {
+			return totalEligibleStake
+		},
+		GetNodeStakedTopUpCalled: func(blsKey []byte) (*big.Int, error) {
+			for shardID, vList := range vInfo {
+				for i, v := range vList {
+					if bytes.Equal(v.PublicKey, blsKey) {
+						return nodesRewardInfo[shardID][i].topUpStake, nil
+					}
 				}
 			}
+			return nil, fmt.Errorf("not found")
+		},
+	}
 
-			remaining := big.NewInt(0).Sub(totalRewardsFirstYear, sumRewards)
-			egldRemaining := remaining.Div(remaining, smallesDivisionERD)
-			maxEGLDDustPerYear := int64(1000)
-			require.True(t, egldRemaining.Int64() < maxEGLDDustPerYear)
-		})
+	args.EconomicsDataProvider.SetRewardsToBeDistributedForBlocks(args.rewardsForBlocks)
+
+	rwd, _ := NewRewardsCreatorV2(args.RewardsCreatorArgsV2)
+
+	var dust *big.Int
+	nodesRewardInfo, dust = rwd.computeRewardsPerNode(vInfo)
+	return nodesRewardInfo, dust
+}
+
+func verifyRewardsComputation(
+	t *testing.T,
+	nbEligiblePerShard uint32,
+	srr SetupRewardsResult,
+	topupStake *big.Int,
+	expectedROI float64,
+) {
+	nodesRewardInfo, _ := computeRewardsAndDust(nbEligiblePerShard, srr, topupStake)
+
+	factor := int64(1000 * 1000)
+	tolerance := 0.01
```

Comment:
the minimum tolerance so the tests will still pass is `0.00977`. maybe use this

Reply:
redundant type conversion on line 1349

Reply:
I would leave the lolerance to 0.01% cause it is easier to read and to explain 
Fixed coversion.